packages/harmony/src/components/button/FollowButton/FollowButton.tsx

```diff
@@ -0,0 +1,122 @@
+
+import type { IconComponent } from 'components/icon'
+import { Flex } from 'components/layout/Flex'
+import { Text } from 'components/text/Text'
+import { useControlled } from 'hooks/useControlled'
+import { IconUserFollowing, IconUserFollow, IconUserUnfollow } from 'icons'
+
+import type { FollowButtonProps } from './types'
+
+const messages = {
+  follow: 'Follow',
+  following: 'Following',
+  unfollow: 'Unfollow'
+}
+
+/**
+ * Special button for following or unfollowing a user.
+ */
+export const FollowButton = (props: FollowButtonProps) => {
+  const {
+    variant = 'default',
+    following = false,
+    onUnfollow,
+    onFollow,
+    size = 'default',
+    ...inputProps
+  } = props
+  const [value, setValueState] = useControlled({
+    componentName: 'FollowButton',
+    controlledProp: following,
+    defaultValue: undefined,
+    stateName: 'following'
+  })
+
+  // Track hover manually to swap text and icon
+  const [isHovering, setIsHovering] = useState(false)
+  const handleMouseEnter = useCallback(() => {
+    setIsHovering(true)
+  }, [setIsHovering])
+  const handleMouseLeave = useCallback(() => {
+    setIsHovering(false)
+  }, [setIsHovering])
+
+  const handleChange = useCallback(
+    (e: ChangeEvent<HTMLInputElement>) => {
+      if (value) {
+        onUnfollow?.()
+      } else {
+        onFollow?.()
+      }
+      setValueState(!value)
+    },
+    [value, setValueState, onFollow, onUnfollow]
+  )
+
+  let Icon: IconComponent | null = IconUserFollow
+  let text = messages.follow
+  if (value && !isHovering) {
+    Icon = IconUserFollowing
+    text = messages.following
+  } else if (value && isHovering) {
+    Icon = IconUserUnfollow
+    text = messages.unfollow
+  }
+
+  const { color, cornerRadius } = useTheme()
+  const textColor =
+    value || isHovering ? color.static.white : color.primary.primary
+  const css: CSSObject = {
+    userSelect: 'none',
+    border: `1px solid ${color.primary.primary}`,
+    borderRadius: variant === 'pill' ? cornerRadius['2xl'] : cornerRadius.s,
+    background:
+      value || isHovering ? color.primary.primary : color.static.white,
+    ':active': {
+      background: color.primary.p500,
+      border: `1px solid ${color.primary.p500}`
+    }
+  }
+
+  return (
+    <>
+      <label onMouseEnter={handleMouseEnter} onMouseLeave={handleMouseLeave}>
+        <input
+          type='checkbox'
+          css={{
+            position: 'absolute',
+            opacity: 0,
+            cursor: 'pointer',
+            height: 0,
+            width: 0
+          }}
+          onChange={handleChange}
+          {...inputProps}
+        />
+        <Flex
+          w={size === 'small' ? 128 : 152}
+          h={size === 'small' ? 28 : 32}
+          direction='row'
+          alignItems='center'
+          justifyContent='center'
+          gap='xs'
+          pv='s'
+          css={css}
+        >
+          <Icon height={18} width={18} css={{ path: { fill: textColor } }} />
```

Comment:
i wonder if we can use the "sizes" and "color" prop available on the icon. looking at default sizes, 18 is not standard... and also looking at available icon colors, "primary" is also not available. maybe we can chat to sammie about

1. using a standard icon size
2. adding primary to the acceptable icon color semantics
3. and maybe finally, allowing an arbitrary color, maybe as "fill" to be an allowed prop to icons?

In the end id like us to do something like:
```
<Icon size='m' color={hovering ? 'staticWhite' : 'primary' />
```

Reply:
Don't want to block on this for too long, so I'm going to merge as-is with a todo and I already started threads in figma with Sammie to confirm. This is still in [beta] so there's room to revise